analyses/survival-analysis/survival-analysis_histology.Rmd

```diff
@@ -0,0 +1,257 @@
+
+1. Univariate analysis 
+- TP53 classifier score (as a continuous variable)
+- EXTEND score (as a continuous variable)
+- Broad histology specified in the color palette file
+- HGG vs. non-HGG
+
+2. Multivariate analysis
+- TP53 classifier score, EXTEND score and HGG group
+- TP53 classifier score, EXTEND score and broad histology specified in the color palette file
+
+3. Plots
+- Bivariate distribution of TP53 classifier score and EXTEND score in density plot stratified by HGAT status
+
+## Setup
+
+#### Packages and functions
+
+Read in set up script.
+
+```{r Set up library}
+library(survival)
+library(dplyr)
+library(ggplot2)
+library(ggpubr)
+```
+
+#### Set up files and directories
+
+Set up output directories. 
+
+```{r Set up directories}
+data_dir <- file.path("..", "..", "data")
+analysis_dir <- file.path("..", "..", "analyses")
+figure_dir <- file.path("..", "..", "figures")
+
+results_dir <- "results"
+plots_dir <- "plots"
+```
+
+Make output directories.
+
+```{r Make output directories}
+if (!dir.exists(results_dir)) {
+  dir.create(results_dir)
+}
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir)
+}
+```
+
+## Declare the direcotry for all the data to be used 
+
+```{r define data path}
+metadata_file <- file.path(data_dir, "pbta-histologies.tsv")
+indep_file <- file.path(data_dir,
+                         "independent-specimens.rnaseq.primary-plus-stranded.tsv")
+tp53_file <- file.path(analysis_dir, 
+                        "tp53_nf1_score/results/tp53_altered_status.tsv")
+# only use stranded since tp53 scores for polyA did not have good accuracy
+tel_stranded_file <- file.path(analysis_dir,
+                         "telomerase-activity-prediction/results/TelomeraseScores_PTBAStranded_FPKM.txt")
+# color palette
+color_file <- file.path(figure_dir, "palettes/broad_histology_cancer_group_palette.tsv")
+```
+
+## Import the metadata
+
+```{r Read in metadata}
+# read in histologies file
+indep_rna <- readr::read_tsv(indep_file)
+
+# read in color file 
+color_bh <- readr::read_tsv(color_file)
+
+# remove cell lines by selecting only solid tissue
+histologies <- readr::read_tsv(metadata_file, guess_max = 10000) %>%
+  dplyr::filter(composition=="Solid Tissue" & tumor_descriptor == "Initial CNS Tumor") %>%
+  dplyr::left_join(color_bh) %>% 
+  dplyr::select(sample_id, broad_histology_display, cancer_group, OS_status, OS_days, PFS_days) %>%
+  distinct()
+
+# read in tp53 scores
+tp53_scores <- readr::read_tsv(tp53_file) %>%
+ select(Kids_First_Biospecimen_ID_DNA, Kids_First_Biospecimen_ID_RNA, sample_id, tp53_score)
+
+# read in telomerase scores
+tel_scores <- readr::read_tsv(tel_stranded_file) %>%
+ mutate(Kids_First_Biospecimen_ID_RNA = SampleID,
+        telomerase_score = NormEXTENDScores) %>%
+ select(Kids_First_Biospecimen_ID_RNA, telomerase_score)
+
+# remove samples without RNA-Seq since both classifiers' results use RNA-Seq
+# independent use primary plus rna-seq samples
```

Comment:
Stating to check my understanding: so the `left_join(histologies)` here means that we're joining the matrices of scores/predictions (which contain all RNA-seq) and keeping all RNA-seq samples until we get to the filter step that uses `indep_rna`.

Reply:
Are we going to be missing metadata (i.e., `OS_status` and `OS_years`) for the non-primary samples with this approach?

Reply:
Since as you pointed out previously, for `histologies`, we filtered to `Initial CNS Tumor` and hence I would think the non-primary participants that only has non primary samples would be missed by this method. Also even if we do not filter to `Initial CNS Tumor` - using the primary independent samples would also filter to samples that are primary. The question is: do we want to keep non-primary?

Reply:
We want to use only primary - but instead of using independent RNA primary-plus sample list, I used `distinct` to directly take only one primary RNA if there are multiple there.